clients/fides-js/src/lib/preferences.ts

```diff
@@ -167,3 +169,140 @@ export const updateConsentPreferences = async ({
+    fides_string: fides.cookie.fides_string || "",
+    tcf_version_hash: fides.cookie.tcf_version_hash || "",
+  };
+
+  // Update cookie with new consent values
+  Object.entries(finalConsent).forEach(([key, value]) => {
+    updatedCookie.consent[key] = value;
+  });
+
+  // If fidesString is provided, update it on the cookie
+  if (fidesString) {
+    updatedCookie.fides_string = fidesString;
+  } else if (consent) {
+    // Generate new fidesString based on updated consent
+    const newNcString = fides.encodeNoticeConsentString(finalConsent);
+    // Preserve other parts of fidesString if they exist
+    if (fides.cookie.fides_string) {
+      const decoded = decodeFidesString(fides.cookie.fides_string);
+      decoded.nc = newNcString;
+      updatedCookie.fides_string = `${decoded.tc || ""}.${newNcString}.${decoded.gpp || ""}`;
+    } else {
+      updatedCookie.fides_string = `.${newNcString}.`;
+    }
+  }
+
+  if (!fides.experience) {
+    return Promise.reject(
+      new Error("Experience must be initialized before updating consent"),
+    );
+  }
+
+  // Prepare consentPreferencesToSave by mapping from finalConsent
+  const consentPreferencesToSave: SaveConsentPreference[] = [];
+
+  // Validate that all notice keys in finalConsent exist in the experience
+  const validNoticeKeys = new Set(
+    fides.experience.privacy_notices?.map((n) => n.notice_key) || [],
+  );
+
+  Object.entries(finalConsent).forEach(([key, value]) => {
+    if (!validNoticeKeys.has(key)) {
+      fidesDebugger(
+        `Warning: Notice key "${key}" does not exist in the current experience`,
+      );
+      // Continue processing other keys
+    }
+
+    const notice = fides.experience?.privacy_notices?.find(
+      (n) => n.notice_key === key,
+    );
+    if (notice) {
+      const historyId = notice.translations?.[0]?.privacy_notice_history_id;
+      if (historyId) {
+        consentPreferencesToSave.push(
+          new SaveConsentPreference(
+            notice,
+            value
+              ? UserConsentPreference.OPT_IN
+              : UserConsentPreference.OPT_OUT,
+            historyId,
+          ),
+        );
+      }
+    }
+  });
+
+  // Get privacy_experience_config_history_id from experience config translations
+  let configHistoryId: string | undefined;
+  if (fides.experience.experience_config?.translations?.length) {
+    configHistoryId =
+      fides.experience.experience_config.translations[0]
+        .privacy_experience_config_history_id;
+  }
+
+  // Call updateConsentPreferences with necessary parameters
+  return updateConsentPreferences({
+    consentPreferencesToSave,
+    privacyExperienceConfigHistoryId: configHistoryId,
+    experience: fides.experience as
+      | PrivacyExperience
+      | PrivacyExperienceMinimal,
+    consentMethod: ConsentMethod.SAVE,
+    options: fides.options,
+    userLocationString: fides.geolocation?.country, // Using country as the location string
```

Comment:
Can we use the fides region string to be consistent with other places we call `updateConsentPreferences`?